File: Server/Tokenizer_V1/Tokenizer_V1/Requests/Templates/CreateTemplateReq.cs
using Microsoft.AspNetCore.Http;
using Tokenizer_V1.Models;

namespace Tokenizer_V1.Requests.Templates
{
    public class CreateTemplateReq
    {
        public int? Id { get; set; }
        public string Name { get; set; }

        public string QrCodeColor { get; set; }
        public string QrCodeBackgroundColor { get; set; }
        public string BackgroundColor { get; set; }
        public string TextColor { get; set; }

        public string CurvedTextTop { get; set; }
        public string CurvedTextBottom { get; set; }
        public int CurvedTextTopOffset { get; set; }
        public int CurvedTextBottomOffset { get; set; }

        public int TokenTypeId { set; get; }
        public int TokenTypeOffset { set; get; }

        public string QrCodeUrl { get; set; }

        public bool UseImage { get; set; }
        public string AltText { get; set; }
        public IFormFile Image { set; get; }
        public int? ProjectId { get; set; }

        public int? CompanyId { get; set; }

    }
}

File: Client/Play It Forward/src/Requests/Templates/CreateTemplateReq.js
class CreateTemplateReq{
    constructor(){
        this.Id = "";
        this.Name = undefined;
        this.Description = undefined;
        this.Amount = undefined;
        this.QrCodeColor = undefined;
        this.QrCodeBackgroundColor = undefined;
        this.BackgroundColor = undefined;
        this.TextColor = undefined;
        this.CurvedTextTop = undefined;
        this.CurvedTextBottom = undefined;
        this.CurvedTextTopOffset = undefined;
        this.CurvedTextBottomOffset = undefined;
        this.TokenTypeId = null;
        this.TokenTypeOffset = undefined;
        this.QrCodeUrl = undefined;
        this.UseImage = undefined;
        this.AltText = undefined;
        this.Image = undefined;
        this.ProjectId = undefined;
        this.CompanyId = undefined;
    }
}
export default CreateTemplateReq;